### PR TITLE
fix: UDP url parsing

### PR DESCRIPTION
**Pull request by alxhotel**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
- As noted in the comments of the code, there is still a bug in Chrome and Firefox which doesn't correctly parse UDP urls.

- The previous code didn't handle the UDP case in the first parsing of the URL:

https://github.com/webtorrent/bittorrent-tracker/blob/2a12b752048d88452d91cd5fa4d8f03ac9131074/client.js#L89

- This PR creates a new function called `parseUrl` in the `common` file to handle the special case for UDP urls, so that the code can be reused wherever is necessary. 

**Which issue (if any) does this pull request address?**
Fixes webtorrent/webtorrent#2066

**Is there anything you'd like reviewers to focus on?**
Let me know if the implementation of `parseUrl` can be improved

### CODE REVIEW

**Review by DiegoRBaquero — APPROVED** [review]

**Comment by jimmywarting** [comment]
How about something like this instead:

```js
str = 'udp://github.com'
url = new URL(str.replace(/^udp:/, 'http:'))

if (str.match(/^udp:/)) {
  Object.defineProperties(url, {
    href: { value: url.href.replace(/^http/, 'udp') },
    protocol: { value: url.protocol.replace(/^http/, 'udp') },
    origin: { value: url.origin.replace(/^http/, 'udp') }
  })
}
```
You will keep it as a URL instances and get support for the rest of the class (serialize to string, instance check, searchParams)

**Comment by alxhotel** [comment]
Neat, thanks @jimmywarting! I have updated the PR.

File: lib/common.js [hunk 1]
```diff
@@ -19,5 +19,31 @@ exports.hexToBinary = function (str) {
...
+// Note: UDP trackers aren't used in the normal browser build, but they are
+// used in a Chrome App build (i.e. by Brave Browser).
+//
+// Bug reports:
+// - Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=734880
+// - Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1374505
+exports.parseUrl = function (str) {
+  const url = new URL(str.replace(/^udp:/, 'http:'))
+
+  if (str.match(/^udp:/)) {
```
Comment (feross) [inline lib/common.js:37]: Nit: Since it's just being used to get a boolean, I prefer to use `/^udp:/.test(str)`.

**Review by feross — APPROVED** [review]
LGTM!

**Pull request merged**